docs/operators/fsInputOperator.md

```diff
@@ -0,0 +1,101 @@
+File Input Operator
+=============
+
+## Operator Objective
+This operator scans a directory for files. Files are then read and split into tuples, which are emitted. The default implementation scans a single directory. The operator is fault tolerant. It tracks previously read files and current offset as part of checkpoint state. In case of failure the operator will skip files that were already processed and fast forward to the offset of the current file. Supports partitioning and changes to number of partitions. The directory scanner is responsible to only accept the files that belong to a partition.
+
+File Input Operator is **idempotent**, **fault-tolerant** and **partitionable**.
+
+## Operator Usecase
+1. Read all files of a directory and then keep scanning it for newly added files.
+
+## Operator Information
+1. Operator location: ***malhar-library***
+2. Available since: ***1.0.2***
+3. Operator state: ***Stable***
+3. Java Packages:
+    * Operator: ***[com.datatorrent.lib.io.fs.AbstractFileInputOperator](https://www.datatorrent.com/docs/apidocs/com/datatorrent/lib/io/fs/AbstractFileInputOperator.html)***
+
+### AbstractFileInputOperator
+This is the abstract implementation that serves as base class for scanning a directory for files and read the files one by one. This class doesn’t have any ports.
```

Comment:
One of the issues that users often ask about is how they can tell which file a tuple belongs to in downstream operators and, a related question, how to tell when one file has ended and new one begun. Discuss a few of ways the user can provide this info and the tradeoffs in each approach, for example (a) Attach the file name to each tuple (b) create a tuple structure that can contain either an actual data tuple or a control tuple indicating begin or end of a file (c) send control tuples on a different port. Some of the examples as well as the Top N Words tutorial have sample
implementations of some of these approaches.